### DISCUSSION TITLE
absolute option "host" disable ?

### DISCUSSION THREAD

**Discussion by stloc** [StenopePHP/Stenope#159] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/4832070/202227448-08e94c1a-e69b-4d63-90e0-303f0dd436fc.png)

result : no host/url insert

![image](https://user-images.githubusercontent.com/4832070/202228606-39c741ce-d85e-4a13-8a35-f58703fc5665.png)

**Comment by ogizanagi** [comment, marked as answer]
Hi @stloc . 

I'm not sure to follow, what would you expect? Why do you need absolute URLs?

Note Stenope does nothing regarding this, so this might be something you need to explore more specifically with [Symfony assets packages](https://symfony.com/doc/current/components/asset.html#absolute-assets-and-cdns), webpack encore config & Twig [`absolute_url`](https://symfony.com/doc/current/reference/twig_reference.html#absolute-url) function.

For instance, here is how we render absolute URLs where required in our website: 
https://github.com/Elao/elao_/blob/3be506d1ce96d34902cd7145c5dadbb0c1f28ed4/templates/blog/article.html.twig#L9-L10

For Webpack, when deploying to a subdirectory and exposing the static files under a base URL, you might need this:

https://github.com/Elao/elao_/blob/3be506d1ce96d34902cd7145c5dadbb0c1f28ed4/webpack.config.js#L12-L15

where `WEBPACK_PUBLIC_PATH` is the base path (_`/static/html` in your case I guess_).

**Reply by stloc** [reply]
Hi,
thanks @ogizanagi 
so what is this option "host" for ?

**Reply by ogizanagi** [reply]
It allows to set the Symfony [router request context](https://symfony.com/doc/current/routing.html#generating-urls-in-commands), to indicate _how_ to generate absolute URLs (which host, base path, etc…) to _use_. But it does not tell to _always generate_ absolute URLs at all. This is your app responsibility.

The simplest way to set this is not using the options btw, but use the [`framework.router.default_uri`](https://symfony.com/blog/new-in-symfony-5-1-simpler-request-context) config with an env var:

https://github.com/Elao/elao_/blob/3be506d1ce96d34902cd7145c5dadbb0c1f28ed4/config/packages/routing.yaml#L8
https://github.com/Elao/elao_/blob/3be506d1ce96d34902cd7145c5dadbb0c1f28ed4/.github/workflows/deploy.yaml#L97

**Reply by stloc** [reply]
thank @ogizanagi for your completes responses !
